src/altinn-app-frontend/src/utils/formComponentUtils.ts

```diff
@@ -412,3 +413,25 @@ export const atleastOneTagExists = (attachments: IAttachment[]): boolean => {
 
   return totalTagCount !== undefined && totalTagCount >= 1;
 };
+
+export function getFieldName(
+  textResourceBindings: ITextResourceBindings,
+  textResources: ITextResource[],
+  language: ILanguage,
+  fieldKey?: string,
+): string {
+  if (fieldKey)
+  {
+    return getTextFromAppOrDefault(`form_filler.${fieldKey}`, textResources, language, null, true);
+  }
+
+  if (textResourceBindings.shortName) {
+    return getTextResourceByKey(textResourceBindings.shortName, textResources);
+  }
+
+  if (textResourceBindings.title) {
```

Comment:
Just a note here. This will be a "breaking" change in the sense that all fields with long title names will suddenly have validation messages that contain the whole title, right?

I'm a little unsure about how real this case is, hopefully, all applications in prod have chosen fairly short and concise titles. Is it worth some time just to investigate if this really is a problem?

Alternative would be to introduce a new major version but would be sweet not to force this on app developers.

Thoughts?

Reply:
Hmm fair point. I think our options are:
1. use only new shortName prop from reedy resources, making the functionality opt-in
  - fallback will be the same, just remove the title condition from this function
3.  Leave as-is
  A. bump to new major version, making the functionality opt-in but with no changes to form layout needed
  B. Bump minor version and push the change to everyone

We don’t have that many apps in production, but we do have a lot of apps underway. I am leaning towards 2A, so we don’t push this out immediately for all apps, and app developers can look over their apps to see if they need to introduce the shortName field anywhere the title field might be too long.
My understanding is that having this kind of information in validation messages is something of a requirement going forward, so making it completely opt-in by option 1 makes it too easy to just skip 🤷‍♀️